packages/devtools_app/lib/src/app_size/app_size_screen.dart

```diff
@@ -93,6 +99,43 @@ class _AppSizeBodyState extends State<AppSizeBody>
     addAutoDisposeListener(_tabController);
   }
 
+  Future<void> maybeLoadAppSizeFiles() async {
+    final queryParams = loadQueryParams();
+    if (queryParams.containsKey(primaryAppSizeFilePropertyName)) {
+      preLoadingData = true;
+      final primaryAppSizeFile = await server.requestPrimaryAppSizeFile(
+          queryParams[primaryAppSizeFilePropertyName]);
+      DevToolsJsonFile secondaryAppSizeFile;
+      if (queryParams.containsKey(secondaryAppSizeFilePropertyName)) {
+        secondaryAppSizeFile = await server.requestSecondaryAppSizeFile(
+            queryParams[secondaryAppSizeFilePropertyName]);
+      }
+
+      if (primaryAppSizeFile != null) {
+        preLoadingData = true;
+        if (secondaryAppSizeFile != null) {
+          controller.loadDiffTreeFromJsonFiles(
+            oldFile: secondaryAppSizeFile,
+            newFile: primaryAppSizeFile,
+            onError: (error) => Notifications.of(context).push(error),
```

Comment:
@pq is this the kind of error case you were seeing where callbacks were issued with a potentially invalid context?
I think we should check whether the context is still valid before pushing the notification. If the page has already been removed from the app then there is no need to notify.

Reply:
checked if mounted